cmd/single-ocsp/main.go

```diff
@@ -0,0 +1,169 @@
+	responderBytes, err := ioutil.ReadFile(responderFileName)
+	if err != nil {
+		return
+	}
+
+	responder, err = x509.ParseCertificate(responderBytes)
+	if err != nil {
+		return
+	}
+
+	// Target certificate
+	targetFileName := c.GlobalString("target")
+	targetBytes, err := ioutil.ReadFile(targetFileName)
+	if err != nil {
+		return
+	}
+
+	target, err = x509.ParseCertificate(targetBytes)
+	if err != nil {
+		return
+	}
+
+	// OCSP template
+	templateFileName := c.GlobalString("template")
+	templateBytes, err := ioutil.ReadFile(templateFileName)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(templateBytes, &template)
+	if err != nil {
+		return
+	}
+
+	// PKCS#11 config
+	pkcs11FileName := c.GlobalString("pkcs11")
+	pkcs11Bytes, err := ioutil.ReadFile(pkcs11FileName)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(pkcs11Bytes, &pkcs11)
+	return
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "admin-revoker"
+	app.Usage = "Revokes issued certificates"
+	app.Version = cmd.Version()
+	app.Author = "Boulder contributors"
+	app.Email = "ca-dev@letsencrypt.org"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "issuer",
+			Usage: "Issuer certificate (DER)",
+		},
+		cli.StringFlag{
+			Name:  "responder",
+			Usage: "OCSP responder certificate (DER)",
+		},
+		cli.StringFlag{
+			Name:  "target",
+			Usage: "Certificate whose status is being reported (DER)",
+		},
+
+		// A partial ocsp.Response.  The remainder will be filled
+		// in from the certificates.
+		//
+		// {
+		//   "Status": 0, // Good
+		//   "ThisUpdate": "2015-08-26T00:00:00Z",
+		//   "NextUpdate": "2016-08-26T00:00:00Z"
+		// }
+		//
+		// {
+		//   "Status": 1, // Revoked
+		//   "ThisUpdate": "2015-08-26T00:00:00Z",
+		//   "NextUpdate": "2016-08-26T00:00:00Z",
+		//   "RevokedAt": "2015-08-20T00:00:00Z",
+		//   "RevocationReason": 1 // Key compromise
+		// }
+		cli.StringFlag{
+			Name:  "template",
+			Usage: "OCSP template file (JSON)",
+		},
+
+		// PKCS#11 config, as defined above
+		//
+		// {
+		//   "Module": "/Library/OpenSC/lib/opensc-pkcs11.so",
+		//   "Token": "Yubico Yubikey NEO CCID",
+		//   "Label": "PIV AUTH key",
+		//   "PIN": "123456"
+		// }
+		cli.StringFlag{
+			Name:  "pkcs11",
+			Usage: "PKCS#11 configuration (JSON)",
+		},
+	}
+
+	app.Action = func(c *cli.Context) {
+		issuer, responder, target, template, pkcs11, err := readFiles(c)
+		cmd.FailOnError(err, "Failed to read files")
+
+		// Instantiate the private key from PKCS11
+		priv, err := pkcs11key.New(pkcs11.Module, pkcs11.Token, pkcs11.PIN, pkcs11.Label)
+		cmd.FailOnError(err, "Failed to load PKCS#11 key")
+
+		// Populate the remaining fields in the template
+		template.SerialNumber = target.SerialNumber
+		template.Certificate = responder
+
+		// Sign the OCSP response
+		responseBytes, err := ocsp.CreateResponse(issuer, responder, template, priv)
+		cmd.FailOnError(err, "Failed to sign OCSP response")
+
+		// Write the OCSP response to stdout
```

Comment:
Since this is binary, it might mess up somebody's terminal if they neglect to redirect stdout to a file. Probably better to require a `-out` param.